cdap-explore/src/main/java/co/cask/cdap/hive/serde/ObjectSerializer.java

```diff
@@ -238,4 +257,23 @@ private Object fromLazyObject(TypeInfo type, Object data) {
         return data.toString();
     }
   }
+
+  private Long toDays(Date obj) {
+    return obj == null ? null : obj.toLocalDate().toEpochDay();
```

Comment:
if obj can be null, it should be annotated accordingly. Alternatively, in the serialize method, you can start off by returning null if the obj is null. That way you don't need this method and can just call obj.toLocalDate().toEpochDay() directly.